server/service/microsoft_mdm.go

```diff
@@ -1004,6 +1009,19 @@ func (svc *Service) storeWindowsMDMEnrolledDevice(ctx context.Context, secTokenM
 		return err
 	}
 
+	err = svc.ds.NewActivity(ctx, nil, &fleet.ActivityTypeMDMEnrolled{
+		HostSerial:  reqDeviceID,
```

Comment:
@marcosd4h I'm fairly confident that this is the serial number, but I'm not 100% sure. Could you sanity check me here? Looking [here](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-mde2/e4e35f28-778f-4e0f-93dd-96e5b3b4637b) I couldn't find documentation about the meaning of each field in `AdditionalContext`

Reply:
this `deviceID` is a unique GUID that is generated by the enrollment client to reference the device on an MDM deployment. It does not contain any serial information, but it is used to univocally track the enrolled device. It think it should be fine to use this value for the context of the activity feed. As a side note, the HWDevID field (a 64char long field) is the one that contains the SMBios serial number information details.

Reply:
@marcosd4h Thank you! as for using `deviceID` the UI copy of the activity item says:

> Mobile device management (MDM) was turned on for a host with serial number SERIAL_NUMBER_HERE (manual)

should we change the copy? or should I use `HWDevID` instead then?

Reply:
I can also load the host and use the serial number we have stored in the `hosts` table, I'm just trying to avoid that if we already have the value, but seems like we don't?

Reply:
> should we change the copy? or should I use HWDevID instead then?

The HWDevID will contain the serial number + other hardware specific field. I'm OK with using this field on the activity UI, the only concern that I've is that the HWDevID is a 64 character long field (Example: `BF2D12A95AE42E47D58462E9A71736CAF13FCCAD3078F340F4D50B371FB2256F`). Will it be OK to display this on the UI?

> I can also load the host and use the serial number we have stored in the hosts table

We don't have an association between enrolled device and the host table atm, because we have not implemented the management protocol yet. So the information will not be there (specially for the automatic enrollment scenarios)

Reply:
Thank you! I added this to the parking lot to double check with Noah and Marko.

> We don't have an association between enrolled device and the host table atm

interesting! curious: what's going to be the foreign key to do the association? I think you mentioned `uuid`? (for Apple it's `uuid`)

Reply:
> interesting! curious: what's going to be the foreign key to do the association?

Great question! Once the management protocol is implemented, we will need to create a new table to maintain the association between HWDeviceID of the MDM-enrolled devices and the Fleet DeviceID. The Fleet DeviceID is already there for programmatic enrollment (fleetd is passing it as part of the auth token), but in the case of Automatic enrollment, we will need to query the [SMBIOS Serial Number](https://learn.microsoft.com/en-us/windows/client-management/mdm/devdetail-csp#extmicrosoftsmbiosserialnumber) through the MDM management protocol to obtain this information and be able to associate a MDM enrolled device with a Fleet DeviceID.

Reply:
>  the only concern that I've is that the HWDevID is a 64 character long field (Example: BF2D12A95AE42E47D58462E9A71736CAF13FCCAD3078F340F4D50B371FB2256F). 

@marcosd4h, is there a reliable specification for the HWDeviceID string that could be used to slice just the serial number (e.g., the first 16 characters are reserved for the serial) or does each hardware vendor do it differently?

Reply:
@marcosd4h thanks, we were just talking with @noahtalerman about this.

The intended UX and the end goal of the activity is for the IT admin to be able to copy this value and do a search in the manage page to find the host. I think if we use `HWDeviceID` or a trimmed version of it they won't be able to do that, right?